formats/json/commonMain/src/kotlinx/serialization/json/JsonElementSerializers.kt

```diff
@@ -117,16 +117,18 @@ private object JsonLiteralSerializer : KSerializer<JsonLiteral> {
             return encoder.encodeInline(value.coerceToInlineType).encodeString(value.content)
         }
 
-        value.longOrNull?.let { return encoder.encodeLong(it) }
+        //use content instead of longOrNull as it can process exponential notation,
+        //and it should be delegated to double when encoding.
+        value.content.toLongOrNull()?.let { return encoder.encodeLong(it) }
 
         // most unsigned values fit to .longOrNull, but not ULong
         value.content.toULongOrNull()?.let {
             encoder.encodeInline(ULong.serializer().descriptor).encodeLong(it.toLong())
             return
         }
 
-        value.doubleOrNull?.let { return encoder.encodeDouble(it) }
-        value.booleanOrNull?.let { return encoder.encodeBoolean(it) }
+        value.content.toDoubleOrNull()?.let { return encoder.encodeDouble(it) }
```

Comment:
`doubleOrNull` simply calls `toDoubleOrNull`, why this change is required?

Reply:
Just to for consistency but its true that this change is not needed at all.